lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -371,6 +374,35 @@ struct CircuitLoweringState {
                        });
   }
 
+  /// Get the sv.verbatim.source op for a filename, if it exists.
+  sv::SVVerbatimSourceOp getVerbatimSourceForFile(StringRef fileName) {
+    llvm::sys::SmartScopedLock<true> lock(verbatimSourcesMutex);
+    auto it = verbatimSourcesByFileName.find(fileName);
+    return it != verbatimSourcesByFileName.end() ? it->second : nullptr;
+  }
+
+  /// Register an sv.verbatim.source op containing the SV implementation for
+  /// some extmodule(s).
+  void registerVerbatimSource(StringRef fileName,
+                              sv::SVVerbatimSourceOp verbatimOp) {
+    llvm::sys::SmartScopedLock<true> lock(verbatimSourcesMutex);
+    verbatimSourcesByFileName[fileName] = verbatimOp;
```

Comment:
This could overwrite. Is that fine?

Also, is it legal for `verbatimOp` to be null here?

The latter would usually be handled with an assert if it is some assumption that isn't supposed to be violated generally.